Dockerfile.ci

```diff
@@ -0,0 +1,92 @@
+#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.20)
+#FROM registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:v1.20.5-202307171904.el8.g844e652 AS builder
+# 1. HACK: package channels don't refresh on the cee image if we're running in CI, so use the CI builder instead of the cee one
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.15 as builder
+
+# 2. HACK: openshift CI doesn't support ENV in image builds, but it does support ARG, so make these ARGs 
+ARG CI_UPSTREAM_VERSION_SANITIZED=main
+ARG CI_CUSTOM_METRICS_AUTOSCALER_OPERATOR_UPSTREAM_COMMIT=HEAD
+ARG CI_CUSTOM_METRICS_AUTOSCALER_OPERATOR_UPSTREAM_VERSION=main
+
+ARG BUILD_VERSION=${CI_UPSTREAM_VERSION_SANITIZED}
+ARG GIT_COMMIT=${CI_CUSTOM_METRICS_AUTOSCALER_OPERATOR_UPSTREAM_COMMIT}
+ARG GIT_VERSION=${CI_CUSTOM_METRICS_AUTOSCALER_OPERATOR_UPSTREAM_VERSION}
+
+
+# 3. HACK: make sure the sources line up with where osbs puts them
+ARG REMOTE_SOURCES_DIR=/src
+ARG REMOTE_SOURCES=.
+COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR/cma-operator/app
+
+WORKDIR $REMOTE_SOURCES_DIR/cma-operator/app
+
+# 4. HACK: osbs feeds in these other sources, there isn't a CI equivalent, so do it here as part of the multistage build
+RUN git clone https://github.com/kubernetes-sigs/controller-tools $REMOTE_SOURCES_DIR/controller-tools/app && touch $REMOTE_SOURCES_DIR/controller-tools/cachito.env
+RUN git clone https://github.com/golang/mock $REMOTE_SOURCES_DIR/mockgen/app && touch $REMOTE_SOURCES_DIR/mockgen/cachito.env
+RUN git clone https://github.com/protocolbuffers/protobuf $REMOTE_SOURCES_DIR/protobuf/app && touch $REMOTE_SOURCES_DIR/protobuf/cachito.env && rm -rf ./$REMOTE_SOURCES_DIR/protobuf/app/examples
+RUN touch ../cachito.env
```

Comment:
I tried to stay as close to what is happening in [midstream](https://gitlab.cee.redhat.com/custom-metrics-autoscaler/cpaas-midstream/-/blob/8a27f6762cbda87226a4b096e3349fbea93f639e/distgit/containers/custom-metrics-autoscaler-operator/Dockerfile.in) but we obviously don't have access to `cachito` or the specific versions these tools are pinned to in midstream, so I was using their default branches.